File: backend/propmanage/src/main/java/app/entity/dto/ApartamentoDTO.java
package app.entity.dto;

import app.entity.StatusApartamento;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class ApartamentoDTO {
    private long id;
    private int apnum;
    private StatusApartamento status;

    public ApartamentoDTO(long id, int apnum, StatusApartamento status) {
        this.id = id;
        this.apnum = apnum;
        this.status = status;
    }
}

File: frontend/propmanage/src/dto/ApartamentoDTO.ts
export interface ApartamentoDTO {
    id: number;
    apnum: number;
    status: StatusApartamento;
}

export enum StatusApartamento {
    OCUPADO = 'OCUPADO',
    VAGO = 'VAGO', 
    MANUTENCAO = 'MANUTENCAO'
}